### PR TITLE
UX - Expandable Search Patient By Any Demographics  - Take 2

### PR DESCRIPTION
**Pull request by zbig01**
Fixed unexpected closing of Search Box issue.

Confirmed that it works as advertised on iPad, iPhone (iOS 12), Android V9, Kindle (FireOS 5.3.6.4)

Patient Finder now responsive up to Smartphone viewport size without superfluous icons

Menu on Vertical tabs more user friendly

RTL adjustments to vertical menu

Please do not forget to run `npm run build`

below added by bradymiller:
Up For Grabs demo for this PR is at: https://www.open-emr.org/wiki/index.php/Development_Demo#Zeta_-_Up_For_Grabs_Demo

### CODE REVIEW

**Comment by bradymiller** [comment]
Up For Grabs demo for this PR is at: https://www.open-emr.org/wiki/index.php/Development_Demo#Zeta_-_Up_For_Grabs_Demo

**Comment by sjpadgett** [comment]
Sorry, but the primary purpose of the button is to open finder and secondary is if any search is populated then run the search and open finder.
The whole reason I took time to add the search any was to help make transition to tabs easier for our long time frame users(same with popup's and dialog). On the other hand the click icon to open finder is a long time tab feature that many users will become very annoyed if we take that feature away. The way I designed it took both sets of users in mind and was the balance between the two Ui's.
With my clinic and a couple others I receive feedback on 502 use, they love and primarily use the search any now. So messing with either is at our peril :)

**Comment by bradymiller** [comment]
hi @zbig01 ,
I also agree with @sjpadgett that the primary function of the search button should be simply opening the search gui. Requiring 2 clicks in addition to needing to wait for the field to open up (albeit I admit it is a really neat thing to watch :) ) seems to be straying from that purpose. Also odd that an empty search is throwing an error rather than just going to the search gui. This is tough because you are trying to do 2 things with one button. Would it make more sense to add another element/button/mechanism(maybe just on smaller screens) that shows/hides the search bar and keep the search icon focused on opening the search gui.

**Comment by zbig01** [comment]
@bradymiller 

In the good old days, in the traditional Frame based view most users were using the Any Demographics text box to do their search. It was a very useful feature especially in the pre Patient Finder days but needed some getting used to.

In its original implementation in OpenEMR 5.0 this feature was removed in the tab based view and the Search button was just used as a link to open the Patient Finder page. In a production environment the user would land on the Patient Finder page showing an unfiltered list of all patients. The user would then have to use the Search box on that page to filter through that list to find the patient. 

The addition of Search by any demographics text box in the tab view re-introduces this very useful feature for tab based layouts. This however has created a situation where the same action - clicking of the button, can result in two different results producing either a filtered or unfiltered list in the Patient Finder page depending on whether or not the Search by any demographics text box contains a search string. 

With this revision of the Search by any demographics text box - the user actions are more streamlined and logical - Button clicked, Search by any demographics text box revealed, the placeholder gives a new user a clue as to the purpose and scope of the search box, natural reaction would be to enter search string and click. User now redirected to the Patient Finder page showing the filtered result. 

What is the purpose of letting the user get to the Patient Finder page by clicking the button and then using a Search box with far less functionality, when the search by any demographics text box gives one the ability to perform a broader search and has the added advantage of retrieving a filtered set of results?

The majority of future users have not seen this feature in action. I suspect that when they are confronted with this search button, on first time use they would read the placeholder text and immediately realize that this search box has added functionality. In a short period of time they would just use this feature rather than the search box on the Patient Finder page :)

The alert that is generated by submitting an empty string in the Search by any demographics text box  is intentional and is used to further reinforce the 'correct' use of that feature :))

**Comment by bradymiller** [comment]
To clarify,
Does the search field at the top right of Patient Finder page not use same algorithm as the search field above tabs?

**Comment by zbig01** [comment]
@bradymiller 

The short answer is no.

A more detailed answer is: 

If any non empty search string is entered in the search any text box then this SQL statement 
```
"SELECT field_id FROM layout_options WHERE form_id = 'DEM' AND field_id not like ? AND uor !=0",
        array('em\_%')
```
in 
`/interface/main/finder/dynamic_finder_ajax.php `

will allow searching  through 75 columns instead of 5 columns - name , DOB, SS no, Home phone and external id

**Comment by sjpadgett** [comment]
Yes and no. Different query but same reporting engine. These are two very different searches where the finder is more granular search types available. Run same query term in both and see difference in results. However i've tested this on large populated production sites where the true difference truly shines.

The reason for not a top search in v5 is most likely due to it is not easy to emulate the finders search box outside of the finders api. The reason it was fairly easily done for any search is because the search query stands on it's own and is not shared between two search engines. 

Primarily openemr is a desktop app. Scaling to tablets can be useful but beyond that, it is just 'kind of' useful. I love the term 'Modern' as applied to user interfaces. It harkens me back to the days when modern meant scrolling banners, pop out inputs and every manner of gizmo one could think up. Modern should be replaced by Good user experience. This means clean, no frills, clear navigation and generally down to business ui. Even our theming is counter intuitive. Many themes have dark backgrounds with dark text with no real contrast making for a squinting eye experience. Three themes, that's it, in my opinion.

This new behavior is odd in function and too be honest, distracting with no real purpose/reason why it is necessary. If the static search is somehow hindering responsiveness on small devices then hide it or collapse based on media types audience. Plus, even the search box styling is out of place. 

In the end, I have tried to be tactful concerning this PR but I've spent many hours considering and designing this feature. I simply don't want the behavior changed.
